pkg/scheduler/profile/profile.go

```diff
@@ -55,13 +54,13 @@ func NewMap(cfgs []config.KubeSchedulerProfile, r frameworkruntime.Registry, rec
 	v := cfgValidator{m: m}
 
 	for _, cfg := range cfgs {
-		if err := v.validate(cfg); err != nil {
```

Comment:
moving validate after newProfile looks weird… is newProfile mutating cfg now?

Reply:
> is newProfile mutating cfg now?

Yes, the multiPoint expansion logic occurs at the instantiation of the framework. This happens here because we need to try casting each MultiPoint plugin to every other plugin type per the design we settled on. This is done on individual Profiles.

After that expansion, the `validate()` call does various cross-Profile checks to ensure the entire config is valid (for example, no conflicting QueueSort plugins are enabled across Profiles). 

As part of this, some checks from `validate()` actually moved in to `newProfile` to still fail at the expected time. There are tests for these errors that we made sure to keep in tact

Reply:
> > is newProfile mutating cfg now?

> Yes,

Sorry, to be more clear this isn't *modifying* the cfg. It's expanding cfg into a framework. The rest of my comment still holds, we are just doing some of these checks on the framework to avoid modifying the cfg